src/WebJobs.Script.WebHost/Middleware/AppServiceHeaderFixupMiddleware.cs

```diff
@@ -29,7 +29,11 @@ public async Task Invoke(HttpContext httpContext)
 
             if (httpContext.Request.Headers.TryGetValue(ForwardedProtocolHeader, out value))
             {
-                httpContext.Request.Scheme = value.FirstOrDefault();
+                var scheme = value.FirstOrDefault();
+                if (scheme != null)
```

Comment:
Could we end up with an empty string here as well?

Reply:
as NickDarvey mentioned, an empty header value is a malformed header. TryGetValue returns false if the header is passed as empty from the client, so I was assuming aspnet will do validation on these things before passing the request to the application. `!string.IsNullOrEmpty()` won't hurt though, even if technically not valid per https://tools.ietf.org/html/rfc7230#section-3.2

Reply:
Makes sense, just wanted to make sure it's something we had validated/thought about.